### PR TITLE
reduce dependencies and code complexity

### PR DESCRIPTION
**Pull request by eddiemonge**
Remove a few dependencies

Reduce some code complexities and general code cleanup

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -11,21 +11,36 @@
 'use strict';
 
 var fs = require('fs');
-var _ = require('lodash');
-var pathExists = require('path-exists');
+var isFunction = require('lodash.isfunction');
```
Comment (sindresorhus) [inline index.js]: You can just do `typeof x === 'function'`. It's what that does.

**Comment by sindresorhus** [comment]
I think we should ES2015ify this module. That would let us remove the last dependency and simplify the code.

**Comment by eddiemonge** [comment]
ES2015ifing it would mean dropping support for node < 4, which I'm ok with but doesn't seem to be super necessary.

File: index.js (hunk 1 above)
Comment (eddiemonge) [inline index.js]: it looked like it had done other things as well but i think it was mostly for edge cases.

**Comment by SBoudrias** [comment]
I'd wait until 0.12 is officially unsupported.

**Comment by sindresorhus** [comment]
That's 2016-12-31 though. Not far away.

**Comment by SBoudrias** [comment]
Feel free to send a PR to remove last dep and drop 0.12 totally :)

**Pull request merged**